test/unit_test/runtime/magnitude_test.cpp

```diff
@@ -129,6 +135,69 @@ TEST_CASE("make_ratio performs prime factorization correctly")
   }
 }
 
+TEST_CASE("magnitude converts to numerical value")
+{
+  SECTION("Positive integer powers of integer bases give integer values")
+  {
+    constexpr auto mag_412 = as_magnitude<412>();
+    check_same_type_and_value(mag_412.value<int>, 412);
+    check_same_type_and_value(mag_412.value<std::size_t>, std::size_t{412});
+    check_same_type_and_value(mag_412.value<float>, 412.0f);
+    check_same_type_and_value(mag_412.value<double>, 412.0);
+  }
+
+  SECTION("Negative integer powers of integer bases compute correct values")
+  {
+    constexpr auto mag_0p125 = as_magnitude<ratio{1, 8}>();
+    check_same_type_and_value(mag_0p125.value<float>, 0.125f);
+    check_same_type_and_value(mag_0p125.value<double>, 0.125);
+  }
+
+  SECTION("pi to the 1 supplies correct values")
+  {
+    constexpr auto pi = pi_to_the<1>();
+    check_same_type_and_value(pi.value<float>, std::numbers::pi_v<float>);
+    check_same_type_and_value(pi.value<double>, std::numbers::pi_v<double>);
+    check_same_type_and_value(pi.value<long double>, std::numbers::pi_v<long double>);
+  }
+
+  SECTION("pi to arbitrary power performs computations in most accurate type at compile time")
+  {
+    constexpr auto pi_cubed = pi_to_the<3>();
+
+    auto cube = [](auto x) { return x * x * x; };
+    constexpr auto via_float = cube(std::numbers::pi_v<float>);
+    constexpr auto via_long_double = static_cast<float>(cube(std::numbers::pi_v<long double>));
+
+    constexpr auto pi_cubed_value = pi_cubed.value<float>;
+    REQUIRE(pi_cubed_value != via_float);
```

Comment:
I think the result is implementation-dependent.

Reply:
How so?  In that `long double` could be the same as `float` on some platforms?

Reply:
Theoretically, since their format is left unspecified.

Reply:
OK; I guarded this whole test behind an `if constexpr` check.